src/libasr/codegen/llvm_utils.cpp

```diff
@@ -373,6 +397,146 @@ namespace LFortran {
+        llvm::Value* current_end_point = LLVM::CreateLoad(*builder,
+                                        get_pointer_to_current_end_point(list));
+        llvm::AllocaInst *i = builder->CreateAlloca(pos_type, nullptr);
+        LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 0)), i);
+        llvm::Value* tmp = nullptr;
+
+        /* Equivalent in C++:
+         * int i = 0;
+         * while(list[i] != item && end_point > i) {
+         *     i++;
+         * }
+         *
+         * if (i == end_point) {
+         *    std::cout << "The list does not contain the element";
+         * }
+         */
+
+        llvm::BasicBlock *loophead = llvm::BasicBlock::Create(context, "loop.head");
+        llvm::BasicBlock *loopbody = llvm::BasicBlock::Create(context, "loop.body");
+        llvm::BasicBlock *loopend = llvm::BasicBlock::Create(context, "loop.end");
+
+        // head
+        llvm_utils->start_new_block(loophead);
+        {
+            llvm::Value* is_item_not_equal = nullptr;
+            llvm::Value* left_arg = read_item(list, LLVM::CreateLoad(*builder, i), false);
+            if( item_type == ASR::ttypeType::Character ) {
+                is_item_not_equal = llvm_utils->lfortran_str_cmp(left_arg, item,
+                                                             "_lpython_str_compare_noteq",
+                                                             module);
+            } else {
+                is_item_not_equal = builder->CreateICmpNE(left_arg, item);
+            }
+            llvm::Value *cond = builder->CreateAnd(is_item_not_equal,
+                                                   builder->CreateICmpSGT(current_end_point,
+                                                    LLVM::CreateLoad(*builder, i)));
+            builder->CreateCondBr(cond, loopbody, loopend);
+        }
+
+        // body
+        llvm_utils->start_new_block(loopbody);
+        {
+            tmp = builder->CreateAdd(
+                        LLVM::CreateLoad(*builder, i),
+                        llvm::ConstantInt::get(context, llvm::APInt(32, 1)));
+            LLVM::CreateStore(*builder, tmp, i);
+        }
+        builder->CreateBr(loophead);
+
+        // end
+        llvm_utils->start_new_block(loopend);
+
+
+        llvm::Function *fn = builder->GetInsertBlock()->getParent();
+        llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
+        llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
+        llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
+
+        llvm::Value* cond = builder->CreateICmpEQ(
+                              LLVM::CreateLoad(*builder, i), current_end_point);
+        builder->CreateCondBr(cond, thenBB, elseBB);
+        builder->SetInsertPoint(thenBB);
+        {
+            // TODO: Allow runtime information like the exact element which is
```

Comment:
This is simple to do but I would do it in a subsequent PR.

Reply:
Thank you for fixing the Exception part. Yup, we can do in the subsequent PR.